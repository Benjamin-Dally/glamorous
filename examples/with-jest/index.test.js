import {matcher, serializer} from 'jest-glamor-react'
import React from 'react'
import {shallow, render, mount} from 'enzyme'
import {Wrapper, Title} from './index'

test('enzyme', () => {
  const ui = (
    <Wrapper>
      <Title>Hello World, this is my first glamor styled component!</Title>
    </Wrapper>
  )

  expect(shallow(ui)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
  expect(mount(ui)).toMatchSnapshotWithGlamor(`enzyme.mount`)
  expect(render(ui)).toMatchSnapshotWithGlamor(`enzyme.render`)
})
