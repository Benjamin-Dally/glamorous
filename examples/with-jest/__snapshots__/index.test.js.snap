// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme.mount 1`] = `
.css-1otybxc,
[data-css-1otybxc] {
  padding: 4em;
  background: papayawhip;
}

.css-1tnuino,
[data-css-1tnuino] {
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
}

<glamorous(section)>
  <section
    className="css-1otybxc"
  >
    <glamorous(h1)>
      <h1
        className="css-1tnuino"
      >
        Hello World, this is my first glamor styled component!
      </h1>
    </glamorous(h1)>
  </section>
</glamorous(section)>
`;

exports[`enzyme.render 1`] = `
.css-1otybxc,
[data-css-1otybxc] {
  padding: 4em;
  background: papayawhip;
}

.css-1tnuino,
[data-css-1tnuino] {
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
}

<section
  class="css-1otybxc"
>
  <h1
    class="css-1tnuino"
  >
    Hello World, this is my first glamor styled component!
  </h1>
</section>
`;

exports[`enzyme.shallow 1`] = `
.css-1otybxc,
[data-css-1otybxc] {
  padding: 4em;
  background: papayawhip;
}

<section
  className="css-1otybxc"
>
  <glamorous(h1)>
    Hello World, this is my first glamor styled component!
  </glamorous(h1)>
</section>
`;
